sakura_core/util/StaticType.h

```diff
@@ -96,39 +96,39 @@ class StaticVector{
 
 //! ヒープを用いない文字列クラス
 //2007.09.23 kobake 作成。
-template <class CHAR_TYPE, int N_BUFFER_COUNT>
+template <int N_BUFFER_COUNT>
 class StaticString{
 private:
-	typedef StaticString<CHAR_TYPE,N_BUFFER_COUNT> Me;
+	using Me = StaticString<N_BUFFER_COUNT>;
 public:
 	static const int BUFFER_COUNT = N_BUFFER_COUNT;
 public:
 	//コンストラクタ・デストラクタ
 	StaticString(){ m_szData[0]=0; }
-	StaticString(const CHAR_TYPE* rhs){ if(!rhs) m_szData[0]=0; else wcscpy(m_szData,rhs); }
+	StaticString(const WCHAR* rhs){ if(!rhs) m_szData[0]=0; else wcscpy(m_szData,rhs); }
 
 	//クラス属性
 	size_t GetBufferCount() const{ return N_BUFFER_COUNT; }
 
 	//データアクセス
-	CHAR_TYPE*       GetBufferPointer()      { return m_szData; }
-	const CHAR_TYPE* GetBufferPointer() const{ return m_szData; }
-	const CHAR_TYPE* c_str()            const{ return m_szData; } //std::string風
+	WCHAR*       GetBufferPointer()      { return m_szData; }
+	const WCHAR* GetBufferPointer() const{ return m_szData; }
+	const WCHAR* c_str()            const{ return m_szData; } //std::string風
 
 	//簡易データアクセス
-	operator       CHAR_TYPE*()      { return m_szData; }
-	operator const CHAR_TYPE*() const{ return m_szData; }
-	CHAR_TYPE At(int nIndex) const{ return m_szData[nIndex]; }
+	operator       WCHAR*()      { return m_szData; }
+	operator const WCHAR*() const{ return m_szData; }
+	WCHAR At(int nIndex) const{ return m_szData[nIndex]; }
 
 	//簡易コピー
-	void Assign(const CHAR_TYPE* src){ if(!src) m_szData[0]=0; else wcscpy_s(m_szData,_countof(m_szData),src); }
-	Me& operator = (const CHAR_TYPE* src){ Assign(src); return *this; }
+	void Assign(const WCHAR* src){ if(!src) m_szData[0]=0; else wcscpy_s(m_szData,_countof(m_szData),src); }
+	Me& operator = (const WCHAR* src){ Assign(src); return *this; }
 
 	//各種メソッド
 	int Length() const { return static_cast<int>(auto_strnlen(m_szData, BUFFER_COUNT)); }
 
 private:
-	CHAR_TYPE m_szData[N_BUFFER_COUNT];
+	WCHAR m_szData[N_BUFFER_COUNT];
```

Comment:
指摘じゃありませんが、こうしとくと若干見映えがよくなります。

```suggestion
	std::array<CHAR_TYPE, BUFFER_COUNT> m_szData{};
```